docs/throttle/README.md

```diff
@@ -0,0 +1,33 @@
+# MinIO Server Throttling Guide [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io) [![Docker Pulls](https://img.shields.io/docker/pulls/minio/minio.svg?maxAge=604800)](https://hub.docker.com/r/minio/minio/)
+
+MinIO server allows two ways to throttle incoming requests
+
+- limiting the number of active requests allowed across the cluster
+- limiting the total wait duration for reach request in the queue
+
+These values are enabled using environment variables only.
+
+## Configure connection limit
+If you have traditional spinning (hdd) drives, some applications with high concurrency might require MinIO cluster to be tuned such that to avoid random I/O on the drives. The way to convert high concurrent I/O into a sequential I/O is by reducing the number of concurrent operations allowed per cluster. This allows MinIO cluster to be operationally resilient to such workloads, while also making sure the drives are at optimal efficiency and responsive.
+
+Example: Limit MinIO cluster to accept at max 1600 simultaneous S3 API requests.
+```sh
+export MINIO_API_CONN_MAX=1600
+export MINIO_ACCESS_KEY=your-access-key
+export MINIO_SECRET_KEY=your-secret-key
+minio server http://server{1...8}/mnt/hdd{1...16}
+```
+
+> NOTE: Setting MINIO_API_CONN_MAX=0 means unlimited and that is the default behavior. These values need to be set based on your deployment requirement and application.
+
+## Configure connection (wait) deadline
+This value works in conjunction with max connection setting, setting this value allows for long waiting requests to quickly timeout whenever there is no slot available to perform the request. Allows for reducing pile up of waiting connections when clients are not configured with better timeouts. Default wait time is *3 minutes* if MINIO_API_CONN_MAX is enabled this needs to be tuned as per applications needs, in our testing *3 minutes* wait time is sufficient for most workloads.
```

Comment:
```suggestion
This value works in conjunction with max connection setting, setting this value allows for long waiting requests to quickly time out when there is no slot available to perform the request. 

This will reduce the pileup of waiting requests when clients are not configured with timeouts. Default wait time is *3 minutes* if MINIO_API_CONN_MAX is enabled. This may need to be tuned to your application needs.
```